src/spatial/flatten.jl

```diff
@@ -0,0 +1,73 @@
+using DiffEqJump, DiffEqBase, LightGraphs
+
+rates = [0.1, 1.0]
+netstoch = [[1 => -1, 2 => -1, 3 => 1], [1 => 1, 2 => 1, 3 => -1]]
+reactstoch = [[1 => 1, 2 => 1], [3 => 1]]
+
+graph = grid((2,2))
+num_species = 3
+num_nodes = nv(graph)
+end_time = 1.0
+tspan = (0.0, end_time)
+u0 = ones(Int, num_species, num_nodes)
+alg = Direct()
+
+hopping_rate = 1.0
+hopping_constants = Vector{Matrix{Float64}}(undef, num_nodes) # TODO QUESTION is this how we want to accept hopping constants? 
+for site in 1:num_nodes
+    hopping_constants[site] = hopping_rate*ones(num_species, DiffEqJump.num_neighbors(graph, site))
+end
+
+#TODO QUESTION should accept variable rx rates?
```

Comment:
Leave variable rx rates for a future PR? Okay.

Reply:
Update: I did add support for variable reaction rates in flattening. It makes more sense, because each flattened reaction gets its own rates after flattening anyway. So may as well accept variable rates as input.